### PR TITLE
Switch to a max counterparty's `dust_limit_satoshis` constant

### PR DESCRIPTION
**Pull request by ariard**
This is a more conservative revamp of #575, see discussion there for rational. Contrary to what we previously discussed we don't have a risk of channel closure triggered by third-party as this check is enforced at channel opening. If our counterparty announces a `dust_limit_satoshis` above 660 sats, we halt the opening.

### CODE REVIEW

File: lightning/src/ln/channel.rs
Comment (TheBlueMatt) [inline lightning/src/ln/channel.rs:447]: Can you add some docs on how you derived the 330 part of this value? In Bitcoin Core I don't think there is a "standard p2wsh output dust threshold".

**Review by TheBlueMatt — COMMENTED** [review]
Concept ACK

**Comment by TheBlueMatt** [comment]
Note that `ln::channel::tests::channel_reestablish_no_updates` and `ln::channel::tests::test_holder_vs_counterparty_dust_limit` are currently failing.

**Comment by ariard** [comment]
Fixed test at 3f534f8 by adding a default HOLDER_DUST_LIMIT_SATOSHIS, replacing the previous `derive_holder_dust_limit_satoshis`.

As this getter is function of current feerate, it doesn't guarantee to be compatible anymore with our new upper bound MAX_DUST_LIMIT_SATOSHIS. This might have lead to reject of our own channel opening.

This also removes the config option `min_limit_satoshis` to always requires that counterparty dust limit satoshis is between HOLDER_DUST_LIMIT_SATOSHIS and MAX_DUST_LIMIT_SATOSHIS. 

I think this needs another reviewer beyond Matt.

Note also I'll manually test the new 330 satoshis limit against Core to test that my computations are good. Should be easier now we have a sample.

File: lightning/src/ln/channel.rs
Comment (TheBlueMatt) [inline lightning/src/ln/channel.rs:525]: The error message here shouldn't print anything about fees anymore - the only issue, I think, is if the total channel value is < 330, plus the fee lookup one line up can be dropped.

File: lightning/src/ln/channel.rs
Comment (TheBlueMatt) [inline lightning/src/ln/channel.rs:702]: This should not mention "user specified" anymore - its not user-specified.

**Comment by valentinewallace** [comment]
Hmm, I guess I'm just questioning the concept of this PR because not everyone's gonna be using bitcoind? (If I'm misunderstanding this, could someone outline the rationale more explicitly rather than a pointer to #575 ?) Plus, aren't we supposed to be the "flexible" lightning implementation? :stuck_out_tongue: I think I'm just missing something but just trying to understand.

**Comment by TheBlueMatt** [comment]
> not everyone's gonna be using bitcoin

As to the minimum dust limit, Bitcoin Core's relay rules are effectively consensus for us. People can run alternate nodes, but if your transaction doesn't meet Bitcoin Core's relay rules, probably it wont find its way to a miner, and even if it did the miner would have to be running something other than Bitcoin Core.

Ultimately, this PR is about addressing lightning "dust inflation" - if your counterparty sets the dust limit higher than is necessary, they can send a number of HTLCs, leave them pending, and then close the channel, burning lots of your funds to fee. We assume that lightning counterparties aren't miners largely for this reason, but ideally it wouldn't be so trivial to burn funds.

**Review by valentinewallace — COMMENTED** [review]
I'm ACK 7c25fb0526154b850826c46a9ac1b578023e051b mod fixing CI and addressing the other comments :)

File: lightning/src/ln/channel.rs
Comment (ariard) [reply lightning/src/ln/channel.rs:525]: Upgraded with a new message, right doesn't need feerate lookup anymore.

**Comment by ariard** [comment]
Updated with comments fixed at 0057afa. Main changes since last time is renaming HOLDER_DUST_LIMIT_SATOSHIS to MIN_DUST_LIMIT_SATOSHIS as this effectively a min required or setup by default on both holder/counterparty commitment transactions.

The only bound we don't enforce is MAX_DUST_LIMIT_SATOSHIS on our own commitment transactions, up to the counterparty to do it.

**Comment by TheBlueMatt** [comment]
The full_stack_target fuzz failure here looks separate from the one fixed in #902 and I assume is new in the PR here.

File: lightning/src/ln/channel.rs
Comment (valentinewallace) [inline lightning/src/ln/channel.rs:734]: Since we select the `holder_selected_channel_reserve_satoshis`, could we just ensure that we never select a reserve below `MIN_DUST_LIMIT_SATOSHIS`?

**Review by valentinewallace — COMMENTED** [review]
Just one question but this looks good

File: lightning/src/ln/channel.rs
Comment (ariard) [reply lightning/src/ln/channel.rs:734]: Do you mean instead of returning an API error, rouding up the `channel_reserve_satoshis` with MIN_DUST_LIMIT_SATOSHIS.

I think I prefer the user to swallow the error and having manually to bounce up the channel value instead of us doing it automatically. We might silently encroach on its expected liquidity ready to use and falsify higher application logic like an accounting app...No a strong opinion here.

File: fuzz/src/full_stack.rs
Comment (TheBlueMatt) [inline fuzz/src/full_stack.rs:920]: Why do we not hit this anymore? seems like this implies we now have less coverage?

**Review by TheBlueMatt — APPROVED** [review]
I'll happily fix up the fuzz test after merge, I think we shouldn't hold this up on it. Looks good otherwise.

File: lightning/src/ln/channel.rs
Comment (valentinewallace) [reply lightning/src/ln/channel.rs:734]: It just seem like it doesn't make sense for `get_holder_selected_channel_reserve_satoshis` to ever return a value less than `330`. But, fine to leave that for follow-up

**Review by valentinewallace — APPROVED** [review]

File: lightning/src/ln/channel.rs
Comment (ariard) [reply lightning/src/ln/channel.rs:734]: I don't think our API prevent a user to try a `new_outbound` with less than 330 sat ? And if does so `get_holder_selected_channel_reserve_satoshis` will return the exact value. 

That said there is a TODO to make more sense of `get_holder_selected_channel_reserve_satoshis`. We can address it at that time.

**Pull request merged**
